### ISSUE TITLE
CoD5 - Admin plugin

### ISSUE THREAD

**Issue by Ragonz** [BigBrotherBot/big-brother-bot#391]
Seem to be having an issue with B3 1.12 [^IronPigeon] where the adminplugin doesnt seem to be able to ban people, see log below. Any ideas?

181015 15:18:23 CONSOLE  '1539613103 say;1737747754;2;#######;!ban @167626 Hack'
181015 15:18:23 VERBOSE  u'Queueing event Say : !ban @167626 Hack'
181015 15:18:23 VERBOSE  'Parsing event: Say: AdminPlugin'
181015 15:18:23 DEBUG    u'AdminPlugin: OnSay handle 9:"!ban @167626 Hack"'
181015 15:18:23 DEBUG    u'AdminPlugin: handle command !ban @167626 Hack'
181015 15:18:23 DEBUG    "Storage: getClientsMatching {'id': u'167626'}"
181015 15:18:23 DEBUG    u'Storage: setClient Client<@167626:1918647011|:"Guvergern":None>'
181015 15:18:23 DEBUG    "Storage: setClient data {'time_add': 1539467311, 'password': u'', 'time_edit': 1539613103, 'name': u'Guvergern', 'group_bits': 0, 'ip': u'#######', 'auto_login': 1, 'greeting': u'', 'pbid': u'', 'connections': 2, 'mask_level': 0, 'login': u'', 'guid': u'1918647011', 'id': 167626}"
181015 15:18:23 ERROR    'AdminPlugin: could not parse event Say: not enough arguments for format string'
181015 15:18:23 VERBOSE  "RCON sending (############) 'tell 2 ^0(^2b3^0)^7: ^8[pm]^7 ^7There was an error processing your'"
181015 15:18:23 VERBOSE  'Parsing event: Say: ChatloggerPlugin'
181015 15:18:23 VERBOSE  u'ChatloggerPlugin: ALL, 66088, #######, !ban @167626 Hack'
181015 15:18:23 DEBUG    'ChatloggerPlugin: rowcount: 1, id:2667601'
181015 15:18:23 VERBOSE  'Parsing event: Say: CensorPlugin'
181015 15:18:23 VERBOSE  'Parsing event: Say: SpamcontrolPlugin'
181015 15:18:23 VERBOSE  'Parsing event: Say: ChatPlugin'

**Comment by Ragonz** [comment]
Punkbuster is switched off as its no longer supported on the game and hasnt been by evenbalance for years

Tempban gives the same issues, I seem to remember a while ago on the main B3 site there used to be a code fix as the guids of CoD could be shorter than expected and you could change it to match with the below setting but cannot find that message anymore.

<set name="guid_length">7</set>

**Comment by 82ndab-Bravo17** [comment]
PB does still work with COD5, though it can be difficult for players to find the correct files.

If its shorter guids that are causing your problem add <set name="guid_length">7</set> to the b3 section of your b3.xml, or guid_length: 7 again in the b3 section if you are using the .ini format. Replace the 7 with whatever you find is needed.

I assume from your reply that it works for most players, just not those who have a shorter guid?

**Comment by Ragonz** [comment]
its xml, added     <set name="guid_length">9</set> to it but no dice, tried     <set name="guid_length">7</set> too and still no difference, just says the same error no matter who its tried on.

**Comment by RaistlinRTCW** [comment]
Hi just to be clear the parameter is set to 7 as a couple of our admins are 7 digit guid

 <set name="guid_length">7</set>

I put the bot in debug mode and tried the ban and the following is the excerpt from the log

181018 16:42:58 CONSOLE '1539877378 say;1737747754;13;-Nagged;\x15!ban @167626 hacks'
181018 16:42:58 DEBUG u'AdminPlugin: OnSay handle 9:"!ban @167626 hacks"'
181018 16:42:58 DEBUG u'AdminPlugin: handle command !ban @167626 hacks'
181018 16:42:58 DEBUG "Storage: getClientsMatching {'id': u'167626'}"
181018 16:42:58 DEBUG u'Storage: setClient Client<@167626:1918647011|:"Guvergern":None>'
181018 16:42:58 DEBUG "Storage: setClient data {'time_add': 1539467311, 'password': u'', 'time_edit': 1539877378, 'name': u'Guvergern', 'group_bits': 0, 'ip': u'83.183.0.124', 'auto_login': 1, 'greeting': u'', 'pbid': u'', 'connections': 2, 'mask_level': 0, 'login': u'', 'guid': u'1918647011', 'id': 167626}"
181018 16:42:58 ERROR 'AdminPlugin: could not parse event Say: not enough arguments for format string'

This is the same for any user we attempt to ban, temp ban, the kick doesn't get updated though it does kick them auto temp bans for too many warnings don't ban them all with the same type of error

Another clan running the same version has the same issue on their install/database

**Comment by 82ndab-Bravo17** [comment]
!help ban should give !ban <name> [<reason>]
Are you using the windows compiled version or the python source version of B3?
Can you see in your log which version of the Admin plugin is being loaded, it should show something like
BOT      'Plugin admin (1.34.4 - ThorN, xlr8or, Courgette, Ozon, Fenix) loaded' fairly near the top, at least if you are using the source version.

**Comment by RaistlinRTCW** [comment]
Yep sorry it does give !ban name reason ,as I said it normally in earlier versions also allows you to do it by @dbnumber also instead of name

181018 16:51:02 BOT      'Plugin admin (1.35 - ThorN, xlr8or, Courgette, Ozon, Fenix) loaded'

And at the very top on start

181018 16:51:01 BOT      'www.bigbrotherbot.net (b3) v1.12 [^IronPigeon] [nt32]'
181018 16:51:01 BOT      'Python: 2.7.14 (v2.7.14:84471935ed, Sep 16 2017, 20:19:30) [MSC v.1500 32 bit (Intel)]'
181018 16:51:01 BOT      'Default encoding: ascii'
181018 16:51:01 BOT      'Starting Cod5Parser v1.4.2  

Ragonez is the admin at the provider so will be able to confirm whether the build is compiled windows version or source

**Comment by Ragonz** [comment]
It’s the precompiled exe version

**Comment by 82ndab-Bravo17** [comment]
I've tried with and without pb, with the exe version and with the source version and it works every time for me.

Have you edited the default plugin_admin config file? If so, try with a fresh copy of the default file and just the admin plugin enabled.

If that doesn't work then try replacing all of the files in the install with a fresh download in a fresh folder and see if it works then with just the admin plugin enabled.

**Comment by danielepantaleone** [comment]
> 181018 16:42:58 ERROR 'AdminPlugin: could not parse event Say: not enough arguments for format string'

Looks like some of the string formats in the Admin Plugin configuration file got messed up and B3 is not able to announce the Ban in the game console. Try as @82ndab-Bravo17 said, and use the stock admin plugin configuration file to see if the problem goes away

**Comment by RaistlinRTCW** [comment]
Hi Gents

Sorry been on holiday, we have tried a fresh install and no joy even with just the admin plugin running I am still getting

181026 11:59:13 CONSOLE  '1540551553 say;1737747754;2;-Nagged;\x15!ban @167626 Hack'
181026 11:59:13 DEBUG    u'AdminPlugin: OnSay handle 9:"!ban @167626 Hack"'
181026 11:59:13 DEBUG    u'AdminPlugin: handle command !ban @167626 Hack'
181026 11:59:13 DEBUG    "Storage: getClientsMatching {'id': u'167626'}"
181026 11:59:13 DEBUG    u'Storage: setClient Client<@167626:1918647011|:"Guvergern":None>'
181026 11:59:13 DEBUG    "Storage: setClient data {'time_add': 1539467311, 'password': u'', 'time_edit': 1540551553, 'name': u'Guvergern', 'group_bits': 0, 'ip': u'83.183.0.124', 'auto_login': 1, 'greeting': u'', 'pbid': u'', 'connections': 2, 'mask_level': 0, 'login': u'', 'guid': u'1918647011', 'id': 167626}"
181026 11:59:13 ERROR    'AdminPlugin: could not parse event Say: not enough arguments for format string'

This is very strange so the only thing I changed in the admin config file was the rules to our standard, I can increase the log level and send the info if need be but I am running out of ideas as never had this in any previous install .

**Comment by 82ndab-Bravo17** [comment]
Did you upgrade from a previous version?

**Comment by RaistlinRTCW** [comment]
Hi

As far as I am aware I am not sure if there was a base install that was upgraded, but the server team are going to install in a completely new folder and we will configure again.

**Comment by RaistlinRTCW** [comment]
Hi Bravo 

Admin log 
https://pastebin.com/raw/VuEAb5Db

And b3 log file
https://pastebin.com/raw/TQ8NG0yx

**Comment by 82ndab-Bravo17** [comment]
Please check the "mesages" section in your b3 config file against the section in the default b3.distribution.ini or xml

**Comment by RaistlinRTCW** [comment]
Hi a fresh install of just 1.12 seems to have resolved the issue thanks for all of your help we can now ban or kick people

**Comment by 82ndab-Bravo17** [comment]
Great!

Your're welcome.
